### ISSUE TITLE
RequireThisCheck doesn't see outer classes for anonymous classes

### ISSUE THREAD

**Issue by artembilan** [checkstyle/checkstyle#3041]
When we use properties from the outer class in the anonymous class the report ends up only with `this.`.
For example:

``` java
public class Foo {

    private final String foo = "foo";

    public static void main(String[] args) {
        new Foo().run();
    }

    private void run() {
        Runnable task = new Runnable() {

            @Override
            public void run() {
                System.out.println(foo);
            }

        };
        task.run();
    }

}
```

The report looks like:

``` xml
<file name="Foo.java">
<error line="32" column="52" severity="error" message="Reference to instance variable &apos;foo&apos; needs &quot;this.&quot;." source="com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck"/>
</file>
```

Although, when we move that `Runnable` to the normal inner class:

``` java
private void run() {
    Runnable task = new MyRunnable();
    task.run();
}

private class MyRunnable implements Runnable {

    @Override
    public void run() {
        System.out.println(foo);
    }

}
```

everything looks good:

``` xml
<file name="Foo.java">
<error line="36" column="44" severity="error" message="Reference to instance variable &apos;foo&apos; needs &quot;Foo.this.&quot;." source="com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck"/>
</file>
```

Thanks and sorry if it duplicates something existing already.
##

**Comment by romani** [comment]
latest behaviour is also not right:
```
/var/tmp $ java -jar checkstyle-6.17-SNAPSHOT-all.jar -c Test.xml TestClass.java
Starting audit...
Audit done.

/var/tmp $ cat Test.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <module name="TreeWalker">
        <module name="RequireThis"/>
    </module>
</module>

/var/tmp $ cat TestClass.java 
public class TestClass {

    private final String foo = "foo";

    public static void main(String[] args) {
        new Foo().run();
    }

    private void run() {
        Runnable task = new Runnable() {

            @Override
            public void run() {
                System.out.println(foo);
            }

        };
        task.run();
    }

}
```

**Comment by MEZk** [comment]
@romani 
By dafault validateOnlyOverlapping is set to true. So there should be violations in your example. 
foo is not overlapped by variable or method argument.

**Comment by romani** [comment]
@MEZk , thanks for a hint , I updated by comment.

**Comment by romani** [comment]
Fix is merged
